streams/src/main/java/org/apache/kafka/streams/state/internals/OrderedBytes.java

```diff
@@ -65,4 +65,16 @@ static Bytes lowerRange(final Bytes key, final byte[] minSuffix) {
                 .array()
         );
     }
+
+    static Bytes range(final byte[] prefix, final Bytes key) {
```

Comment:
Please let me know if you want to keep it as is -- which is fine.

But I think we should try to avoid byte array copies as much as possible: right now we first do a byte array for the prefix, in the schema, and then allocate a longer byte array copying the prefix, and then the key bytes. We can, instead, moving this logic directly to the schema where we just create a single array once, and put in the timestamp / sequence / key correspondingly